src/PluginCreator.ts

```diff
@@ -0,0 +1,195 @@
+import * as ts from 'typescript'
+import * as resolve from 'resolve'
+import compareVersions from 'compare-versions'
+
+export type TransformerPlugin = {
+    before?: ts.TransformerFactory<ts.SourceFile>
+    after?: ts.TransformerFactory<ts.SourceFile>
+    afterDeclaration?: ts.TransformerFactory<ts.SourceFile>
+} | ts.TransformerFactory<ts.SourceFile>
+
+export type FactoryType = 'ls' | 'program' | 'opts' | 'checker'
+
+export type PluginConfig = {
+    transform: string
+    type?: FactoryType
+    after?: boolean
+    before?: boolean
+    afterDeclaration?: boolean
+    [options: string]: any;
+} | (ts.PluginImport & {
+    [options: string]: any
+})
+
+export type PluginFactory = {
+    type: 'ls'
+    (ls: ts.LanguageService, options?: PluginConfig): TransformerPlugin
+} | {
+    type?: 'program'
+    (program: ts.Program, options?: PluginConfig): TransformerPlugin
+} | {
+    type: 'opts'
+    (opts: ts.CompilerOptions, options?: PluginConfig): TransformerPlugin
+} | {
+    type: 'checker'
+    (opts: ts.TypeChecker, options?: PluginConfig): TransformerPlugin
+}
+
+export type TransformerHost = ts.LanguageService | ts.Program | ts.CompilerOptions | ts.TypeChecker
```

Comment:
You transfer to `creator.createTransformers(program)` only `program`
Why do you need `TransformerHost`?

Reply:
PluginCreator can be used as separate helper in [rollup-plugin-typescript2](https://github.com/ezolenko/rollup-plugin-typescript2/blob/master/src/ioptions.ts#L26) and other hosts.